uint8 type
uint8 robot
uint16 node